src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -248,8 +249,15 @@ def _copy_previous_run_result(
                     # Thus the root_run_id needs to be the current batch run id.
                     previous_run_info.root_run_id = run_id
                     previous_run_info.parent_run_id = run_id
+
                     # Load previous node run info
                     previous_node_run_infos = resume_from_run_storage.load_node_run_info_for_line(i)
+                    # In storage, aggregation nodes are persisted with filenames similar to regular nodes.
+                    # When reading node run records by filename, we may load unexpected records of aggregation nodes.
```

Comment:
Please add an example folder structure, e.g.
```
node_artifacts/
├─ non-aggregation-node/
│  ├─ 000000000.jsonl
│  ├─ 000000001.jsonl
│  ├─ 000000001=2.jsonl
├─ aggregation-node/
│  ├─ 000000000.jsonl
```
node_artifacts/aggregation-node/000000000.jsonl will be treated as the node_run_info of the first line